### ISSUE TITLE
Unable to play app trailers on Steam with `no-large-media` 

### ISSUE THREAD

**Issue by przemo20** [uBlockOrigin/uBlock-issues#3325]
### Prerequisites

- [X] I verified that this is not a filter list issue. Report any issues with filter lists or broken website functionality in the [uAssets issue tracker](https://github.com/uBlockOrigin/uAssets/issues).
- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] The issue is not present after disabling uBO in the browser.
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

With the `no-large-media` rule added for `store.steampowered.com`, the videos in the screenshot carousel on the Steam store app pages get blocked, and there's no way to unblock them (without removing the added `no-large-media` rule) – if you click on the video, then on the black box (to unlock it), nothing happens.

#### Screenshot

![obraz](https://github.com/user-attachments/assets/86d65be1-ec54-4350-9bea-d6a478628875)

### A specific URL where the issue occurs.

```html
https://store.steampowered.com/app/238960/Path_of_Exile/
```

### Steps to Reproduce

1. Open uBlock Origin Dashboard, switch to the “My rules” tab, add the following temporary rule:
`no-large-media: store.steampowered.com true`, then click on the “Save” button and then the “Commit” button
2. Go back to the uBO “Settings” tab, and make sure that the value next to “Block media elements larger than” is, let's say, not bigger than 2500 KB _(you don't need to check the box next to that option)_
3. Restart the browser _(just to make sure that the new settings get applied correctly)_
4. Go to https://store.steampowered.com/app/238960/Path_of_Exile/
5. Wait for the page to load, then click on the black box (below the game name)

#### Video recording

```html
https://www.youtube.com/watch?v=MI-cpxkHTF0
```

### Expected behavior

The video (and the video player) should load when you click on it.

### Actual behavior

See Description.

On Firefox, I see the following error in the browser console when trying to view the trailer – “DOMException: The media resource indicated by the src attribute or assigned media provider object was not suitable.” and on Chrome – “DOMException: Failed to load because no supported source was found.”

But if you – after the step 3 – select a screenshot from the screenshot carousel, then click on it [the black box] – the image loads fine.

### uBO version

1.59.0

### Browser name and version

Mozilla Firefox 128.0.3 (64-bit)

### Operating System and version

Windows 10 Pro 22H2 (build 19045.4651)

**Comment by przemo20** [comment]
Yeah, blocking them works as intended, though there should still be a way to unlock/load them, just like you can ublock/load large images by clicking on their placeholders (without disabling `no-large-media`).

**Comment by stephenhawk8054** [comment]
Hmm... The images element can be wrapped by using website's `<img>` tag, hence able to have click events. Videos might be able to wrapped by using website's `<video>` tag.

A simpler example would be this: https://interactive-examples.mdn.mozilla.net/pages/tabbed/video.html

However, that won't help in `steampowered.com` case either. The media are lazyloaded and there's no media tags (`img`, `video`...) to be wrapped around

![image](https://github.com/user-attachments/assets/d0f50a96-7649-4c8b-b4e0-dc2bdacf9731)

**Comment by gorhill** [comment]
There is an issue with blocked videos by no-large-element in Firefox, they can't be clicked-to-load.

**Comment by ChaseKnowlden** [comment]
Can't reproduce the issue.

**Comment by przemo20** [comment]
> Can't reproduce the issue.

I slightly revised the “Steps to Reproduce” section (I edited the Step 1 to say that you need to click on the “Save” button first after adding the `no-large-media: store.steampowered.com true` entry before you can hit “Commit” and added step 2 and 3), and added a video recording of the issue as well. I hope this will make it easier to reproduce.

**Comment by ChaseKnowlden** [comment]
![image](https://github.com/user-attachments/assets/df4c4b1a-5b26-40dc-9478-7c2e831422e1)

It now reproduces the issue after following the modified steps.

**Comment by gwarser** [comment]
Right click menu entry is also missing.

**Comment by gorhill** [comment]
The `video` element is added programmatically when clicking the thumbnail, and the code adding it fails because of the `no-large-media-elements` switch, which causes the `video` element to not be seen by uBO's click handler. Not sure this can be addressed.

The workaround is to toggle off no-large-media-elements from the popup panel and force a reload.

**Comment by przemo20** [comment]
Could you maybe instead make/add an option where you can select what kind of `media elements` _(that exceed the specified size)_ we want to block? So for example, I'm fine with **videos** on the Steam store _(since there's an option in their video player to stop videos from autoplaying)_ – I just don't want for **big GIF images** to load automatically _(that some developers overuse in the descriptions of their games)_.

Not sure where you would put this new option:
– would it be a sub-option under “Block media elements larger than” (with options like “Videos (*.mp4; *.webm etc.)”, “Images (*.gif; *.png etc.)”, “Music”)?
– or maybe you would have an option to specify that in the rule on the “My rules” page, like `no-large-media: store.steampowered.com true` (would mean block all media), but e.g. `no-large-media (*.png, *.gif): store.steampowered.com true` (or something like that; would block only the big PNG and GIF images on `store.steampowered.com`)?
– or maybe you could change that somewhere on the `advanced-settings.html` extension page?

It's just a suggestion.

**Comment by gwarser** [comment]
@gorhill why the context menu entry is missing? There are plenty of other images blocked, so content script should be aware of them?

**Comment by gorhill** [comment]
It's there for me:

![a](https://github.com/user-attachments/assets/aabebc17-e517-48da-9067-d83faf0ffd96)

**Comment by gwarser** [comment]
Ok, if I wait long enough it appears, but only once. On subsequent clicks does not appear. Wierd.

**Comment by przemo20** [comment]
I guess this issue can now be closed. Steam upgraded their video player today, and now uBO doesn't delete it with `no-large-media` enabled, and you can play the videos just fine.
